Framework/Crystal/test/IntegratePeakTimeSlicesTest.h

```diff
@@ -97,44 +99,36 @@ class IntegratePeakTimeSlicesTest : public CxxTest::TestSuite {
     boost::shared_ptr<Geometry::Detector> pixelp =
         bankR->getAtXY(PeakCol, PeakRow);
 
-    Geometry::IDetector_const_sptr pix = wsPtr->getDetector(522);
-
     // Now get Peak.
     double PeakTime = 18000 + (PeakChan + .5) * 100;
 
-    Mantid::Kernel::Units::Wavelength wl;
-    Kernel::V3D pos = Kernel::V3D(instP->getSource()->getPos());
-    pos -= instP->getSample()->getPos();
-    double L1 = pos.norm();
-    Kernel::V3D pos1 = pixelp->getPos();
-    pos1 -= instP->getSample()->getPos();
-    double L2 = pos1.norm();
-    double dummy, phi;
-    pos1.getSpherical(dummy, phi, dummy);
-    double ScatAng = phi / 180 * M_PI;
+    const auto &spectrumInfo = wsPtr->spectrumInfo();
+    const auto &detectorInfo = wsPtr->detectorInfo();
+    const auto detID = pixelp->getID();
+    const auto wsIndex = detectorInfo.indexOf(detID);
+    const double L1 = spectrumInfo.l1();
+    double L2 = spectrumInfo.l2(wsIndex);
+    double ScatAng = detectorInfo.twoTheta(wsIndex) / 180 * M_PI;
```

Comment:
L1 and sample position, yes. I think I should leave the rest as-is as the L2 has to be calculated for an individual detector and they anyway need the phi angle which is not supplied by SpectrumInfo.